subprojects/file-watching/src/main/java/org/gradle/internal/watch/registry/impl/DarwinFileWatcherRegistryFactory.java

```diff
@@ -42,6 +45,22 @@ protected FileWatcher createFileWatcher(BlockingQueue<FileWatchEvent> fileEvents
 
     @Override
     protected FileWatcherUpdater createFileWatcherUpdater(FileWatcher watcher) {
-        return new HierarchicalFileWatcherUpdater(watcher);
+        return new HierarchicalFileWatcherUpdater(watcher, DarwinFileWatcherRegistryFactory::validateLocationToWatch);
```

Comment:
Nit: why not simply use a lambda?

Reply:
I found the code too long for a lambda and preferred adding a method.